### PR TITLE
Saving and loading from disk.

### PR DESCRIPTION
**Pull request by RodionShyshkin**
Methods for saving and loading tasks from disk were added in storage, covered by tests and documented.
We can see a new Serialization directory, which contains TaskSerializer (converts TaskEntity to ProtoTask) & TaskDeserializer (vice versa). Also there is a task.proto file which has a protobuf model of TaskEntity & storage.

To check added methods see FullStorage.cpp.

The next step will be separating saving/loading from storage by implementing Persister class, and also adding this feature to API.

### CODE REVIEW

File: src/core/MemoryModel/Storage/FullStorage.cpp [hunk 1]
```diff
@@ -74,3 +80,51 @@ OperationResult FullStorage::RemoveTask(const TaskID &id) {
...
+  FullStorage temp_storage_;
+  for(const auto& task : storage.tasks()) {
+    auto task_ = TaskDeserializer::DeserializeTask(task);
+    if(task_ == nullptr) return OperationResult{ErrorCode::INVALID_TASK};
+    TaskDTO dto_ = TaskDTO::create(0, task_->GetName(), task_->GetLabel(), task_->GetPriority(),
+                                   task_->GetDueTime().GetDate(), task_->GetStatus());
+
+    OperationResult result_of_adding{ErrorCode::NO_ERRORS};
+    if(task.parent_id() == 0) {
+      result_of_adding = temp_storage_.AddTask(dto_);
```
Comment (ValeriiSudakov) [inline src/core/MemoryModel/Storage/FullStorage.cpp]: "Result" will be enough to understand

File: src/core/MemoryModel/Storage/Serialization/TaskDeserializer.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  /*
+   * \brief Converts TaskProto_Priority to Priority.
+   *
+   * @param TaskProrot_Priority
+   *
+   * @return Priority instance.
+   */
+  static Priority         DeserializePriority(const TaskProto_Priority&);
+};
+
```
Comment (ValeriiSudakov) [inline src/core/MemoryModel/Storage/Serialization/TaskDeserializer.h]: Shouldnt it be a virtual methods for mock in tests in future?

File: src/core/MemoryModel/Storage/Serialization/TaskDeserializer.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// Created by rodion on 9/26/20.
+//
+
+#include "TaskDeserializer.h"
+
+std::shared_ptr<TaskEntity> TaskDeserializer::DeserializeTask(const TaskProto &task) {
+  auto task_ = Task::create(task.name(), task.label(),
+                            TaskDeserializer::DeserializePriority(task.priority()),
+                            boost::gregorian::date{task.deadline()});
+  if(!task_.has_value()) return nullptr;
```
Comment (ValeriiSudakov) [inline src/core/MemoryModel/Storage/Serialization/TaskDeserializer.cpp]: why don't you use "Optional" as return value? 
someone, who call this method, will check for nullptr.
Reads better "has_value".

File: src/core/MemoryModel/Storage/Serialization/TaskSerializer.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
+//
+// Created by rodion on 9/25/20.
+//
+
+#include "TaskSerializer.h"
+
+TaskProto TaskSerializer::SerializeTaskWithSubtasks(const TaskEntity &task) {
+  auto* newtask = new TaskProto;
```
Comment (ValeriiSudakov) [inline src/core/MemoryModel/Storage/Serialization/TaskSerializer.cpp]: Raw pointers are taboo, arent they?

**Review by ValeriiSudakov — CHANGES_REQUESTED** [review]

File: src/core/MemoryModel/Storage/Serialization/TaskDeserializer.h (hunk 2 above)
Comment (RodionShyshkin) [reply src/core/MemoryModel/Storage/Serialization/TaskDeserializer.h]: No, because now it's just converter from TaskEntity into TaskProto.

**Review by ValeriiSudakov — APPROVED** [review]

**Pull request merged**